### PR TITLE
skeleton elevator subsystem start

### PR DESCRIPTION
**Pull request by veronayue**
add some functions-- rise, lower,stop, isCloseHighest, isCloseLowest

### CODE REVIEW

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+import xbot.common.command.BaseSubsystem;
+
+
+@Singleton
+public class ElevatorSubsystem extends BaseSubsystem {
+	
+	@Inject
+	public ElevatorSubsystem() {
+		// TODO Auto-generated constructor stub
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Remove this comment

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+	
+	}
+
+	/**
+	 * the elevator will rise by certain speed, the speed may change
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Most functions should be described in a verb-y way:
"Raises the elevator. Power is controlled by a property."

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+
+import xbot.common.command.BaseSubsystem;
+
+
+@Singleton
+public class ElevatorSubsystem extends BaseSubsystem {
+
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:11]: Let's add a DoubleProperty that keeps track of the default elevator power.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+	/**
+	 * the elevator will rise by certain speed, the speed may change
+	 */
+	public void rise(){
+		
+	}
+	
+	
+	/**
+	 * the elevator will fail by certain speed, the speed may change
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: "Lower the elevator.  Power is controlled by a property."

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 5]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	public void lower(){
+	//move when hold the button??
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Remove this comment - the "why" e.g. a button triggering this is not important.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,60 @@
...
+	}
+	
+	public void stop(){
+		
+	}
+	
+	/**
+	 * if it's close to the max height
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Start comments with capital letters.
"Returns true if the elevator is close to its maximum height."

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	boolean isCloseHighest(){
+		return false;
+		}
+	
+	/**
+	 * if it's close to the min height
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: See above

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 8]
```diff
@@ -0,0 +1,60 @@
...
+	 */
+	boolean isCloseLowest(){
+		return false;
+	}
+	
+
+	public void toMaxHeight(){
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Add a verb like "moveToMaxHeight"

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 9]
```diff
@@ -0,0 +1,60 @@
...
+		
+	}
+	public void toMinHeight(){
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: see above

**Review by JohnGilb — CHANGES_REQUESTED** [review]

**Comment by WasabiFan** [comment]
@JohnGilb Did you guys already have a discussion about how this was going to be designed over all? Is this going to be setpoint-oriented, manual control, or a hybrid like 2015?

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 3 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java:11]: Sounds Good!

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 1 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Done.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 2 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Got it!

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 4 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Got it!

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 5 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Okay.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 6 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Okay.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 7 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Okay.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 8 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Okay.

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java (hunk 9 above)
Comment (veronayue) [reply CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Sounds good!

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 10]
```diff
@@ -0,0 +1,61 @@
...
+	}
+	
+	public void stop(){
+		
+	}
+	
+	/**
+	 * Returns true if the elevator is close to its maximum height.
+	 */
+	boolean isCloseHighest(){
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: Let's make this even more descriptive (long names are OK):
"isCloseToMaximumHeight"

File: CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java [hunk 11]
```diff
@@ -0,0 +1,61 @@
...
+		return false;
+		}
+	
+	/**
+	 * Returns true if the elevator is close to its minimum height.
+	 */
+	boolean isCloseLowest(){
```
Comment (JohnGilb) [inline CompetitionTemplate/src/main/java/competition/subsystems/elevator/ElevatorSubsystem.java]: "isCloseToMinimumHeight"

**Review by JohnGilb — CHANGES_REQUESTED** [review]

**Comment by JohnGilb** [comment]
@WasabiFan This will likely be upgraded at some point to be setpoint-based, especially if the motor has to remain active in order to hold at a specific height.

**Review by JohnGilb — APPROVED** [review]

**Pull request merged**